mysql-ballerina/Module.md

```diff
@@ -0,0 +1,391 @@
+version= "0.1.0"
+
+[[platform.java11.dependency]]
+artifactId = "mysql-connector-java"
+version = "8.0.17"
+path = "/path/to/mysql-connector-java-8.0.17.jar"
+groupId = "mysql"
+``` 
+
+### Client
+To access a database, you must first create a
+[mysql:Client](https://docs.central.ballerina.io/ballerinax/mysql/latest/clients/Client) object.
+The examples for creating a MySQL client can be found below.
+
+#### Creating a Client
+This example shows the different ways of creating the `mysql:Client`.
+
+The client can be created with an empty constructor, and thereby, the client will be initialized with the default properties.
+
+```ballerina
+mysql:Client|sql:Error dbClient = new ();
+```
+
+The `dbClient` receives the host, username, and password. Since the properties are passed in the same order as they are defined
+in the `jdbc:Client`, you can pass them without named params.
+
+```ballerina
+mysql:Client|sql:Error dbClient = new ("localhost", "rootUser", "rooPass", 
+                              "information_schema", 3306);
+```
+
+The `dbClient` uses the named params to pass the attributes since it is skipping some params in the constructor.
+Further, the [`mysql:Options`](https://docs.central.ballerina.io/ballerinax/mysql/latest/records/Options)
+property is passed to configure the SSL and connection timeout in the MySQL client.
+
+```ballerina
+mysql:Options mysqlOptions = {
+  ssl: {
+    mode: mysql:SSL_PREFERRED
+  },
+  connectTimeout: 10
+};
+mysql:Client|sql:Error dbClient = new (user = "rootUser", password = "rootPass",
+                              options = mysqlOptions);
+```
+
+Similarly, the `dbClient` uses the named params and it provides an unshared connection pool of the type of
+[sql:ConnectionPool](https://docs.central.ballerina.io/ballerina/sql/latest/records/ConnectionPool)
+to be used within the client.
+For more details about connection pooling, see the [`sql` Module](https://docs.central.ballerina.io/ballerina/sql/latest).
+
+```ballerina
+mysql:Client|sql:Error dbClient = new (user = "rootUser", password = "rootPass",
+                              connectionPool = {maxOpenConnections: 5});
+```
+
+#### Using SSL
+To connect the MySQL database using an SSL connection, you must add the SSL configurations to the `mysql:Options` when creating the `dbClient`.
+For the SSL Mode, you can select one of the modes: `mysql:SSL_PREFERRED`, `mysql:SSL_REQUIRED`, `mysql:SSL_VERIFY_CA`, or `mysql:SSL_VERIFY_IDENTITY` according to the requirement.
+For the key and cert files, you must provide the files in the `.p12` format.
+
+```ballerina
+string clientStorePath = "/path/to/keystore.p12";
+string turstStorePath = "/path/to/truststore.p12";
+
+mysql:Options mysqlOptions = {
+  ssl: {
+    mode: mysql:SSL_PREFERRED,
+    key: {
+        path: clientStorePath,
+        password: "password"
+    },
+    cert: {
+        path: turstStorePath,
+        password: "password"
+    }
+  }
+};
+```
+#### Connection Pool Handling
```

Comment:
This same as SQL, shall we point to SQL API docs?

Reply:
According to @daneshk 's input, He said that it is better to add samples rather than pointing SQL API docs and duplication of samples are ok.

Reply:
Ack